### ISSUE TITLE
hr support

### ISSUE THREAD

**Issue by cittadhammo** [LegNeato/mdbook-typst#13]
hr in markdown are `***` on a new line.

In typst they could be

```
#let hrule = line(length: 100%)
#let hrule = align(center, line(length: 60%))

#hrule
```

https://github.com/typst/typst/discussions/3708

**Comment by LegNeato** [comment]
Sounds good! I can poke at these in a couple of weeks when I get back from vacation unless you get to it first. Thank you for the bug reports and suggestions!

**Comment by cittadhammo** [comment]
Yes, I gave it some thought and started looking at your code. The AI said to implement a custom case to catch any hr in the markdown in there: 

![image](https://github.com/user-attachments/assets/10795b58-200f-4fbf-a2d0-3566167c104f)

with something like:

```
 // Add this case to handle horizontal rules
            (_, Some(ParserEvent::Mdbook(MdbookEvent::MarkdownContentEvent(MdEvent::Start(MdTag::HorizontalRule)))) ) => {
                // Convert hr to Typst
                return Some(ParserEvent::Typst(TypstEvent::Raw(
                    r#"#hr()"#.into(), // Assuming this is how you'd denote a horizontal line in Typst
                )));
            }
```

But I have to say that your code is almost incomprehensible to me plus the layer of that  pullup project which is something else you need to understand (events and what not)

If you have some guidance to start learning how it works, I could have a go, but it looks quite advanced to me. I am used to JS and Node projects.

**Comment by LegNeato** [comment]
Yeah, the code is pretty ugly because 1) it is broken up into reusable parts (maybe too much!) and 2) everything is iterators, which makes it efficient via stream processing but makes the control flow hard to follow.

**Comment by LegNeato** [comment]
It's unfortunate that typst doesn't appear to have a function for this. Seems like an obvious feature to have a semantic rule rather than having to draw one with a line. Perhaps I'll put up a patch to add it.

**Comment by LegNeato** [comment]
This should be fixed in v0.1.7. Let me know if it is not!

Note that there is a new config option for specifying what you want HR to output.
